tools/clang/tools/dxcvalidator/dxcvalidator.cpp

```diff
@@ -191,32 +250,56 @@ uint32_t hlsl::validateWithDebug(
        OptDebugBitcode->Size >= UINT32_MAX))
     return E_INVALIDARG;
 
-  HRESULT hr = S_OK;
+  HRESULT HR = S_OK;
+  HRESULT ValidationStatus = S_OK;
   DxcThreadMalloc TM(DxcGetThreadMallocNoRef());
   try {
-    LLVMContext Ctx;
     CComPtr<AbstractMemoryStream> DiagMemStream;
-    hr = CreateMemoryStream(TM.GetInstalledAllocator(), &DiagMemStream);
-    if (FAILED(hr))
-      throw hlsl::Exception(hr);
-    raw_stream_ostream DiagStream(DiagMemStream);
-    llvm::DiagnosticPrinterRawOStream DiagPrinter(DiagStream);
-    PrintDiagnosticContext DiagContext(DiagPrinter);
-    Ctx.setDiagnosticHandler(PrintDiagnosticContext::PrintDiagnosticHandler,
-                             &DiagContext, true);
-    std::unique_ptr<llvm::Module> DebugModule;
-    if (OptDebugBitcode) {
-      hr = ValidateLoadModule((const char *)OptDebugBitcode->Ptr,
```

Comment:
Why has this been removed?  We should be passing the loaded debug module through to runValidation, right?

Reply:
Nevermind, it's just missing from the `DxcValidatorFlags_ModuleOnly` case, I'll make a separate comment.

Reply:
Ok, I think this original structure should be restored so that a debug module may be passed down to a `validateWithOptModules` function that will apply the hash when applicable.

Reply:
Recovered debug module.